In [1]:
from sklearn.datasets import load_iris
import pandas as pd
import requests

# Load the Iris dataset
iris = load_iris()

# Create a DataFrame
df = (pd.DataFrame(data=iris.data, columns=iris.feature_names)
      .sample(frac=1)
      .reset_index(drop=True))

# Make predictions
url = 'http://127.0.0.1:5000/predict'
for i in range(df.shape[0]):
    data = {'features': df.loc[i].to_list()}
    response = requests.post(url, json=data)
    print(data, response.json())

{'features': [6.3, 2.5, 5.0, 1.9]} {'prediction': 'virginica'}
{'features': [4.8, 3.4, 1.6, 0.2]} {'prediction': 'setosa'}
{'features': [5.0, 3.0, 1.6, 0.2]} {'prediction': 'setosa'}
{'features': [5.6, 2.8, 4.9, 2.0]} {'prediction': 'virginica'}
{'features': [5.5, 2.4, 3.8, 1.1]} {'prediction': 'versicolor'}
{'features': [5.3, 3.7, 1.5, 0.2]} {'prediction': 'setosa'}
{'features': [5.5, 2.3, 4.0, 1.3]} {'prediction': 'versicolor'}
{'features': [6.3, 2.9, 5.6, 1.8]} {'prediction': 'virginica'}
{'features': [4.5, 2.3, 1.3, 0.3]} {'prediction': 'setosa'}
{'features': [5.7, 3.0, 4.2, 1.2]} {'prediction': 'versicolor'}
{'features': [6.4, 2.8, 5.6, 2.1]} {'prediction': 'virginica'}
{'features': [6.9, 3.2, 5.7, 2.3]} {'prediction': 'virginica'}
{'features': [5.7, 2.9, 4.2, 1.3]} {'prediction': 'versicolor'}
{'features': [5.9, 3.0, 5.1, 1.8]} {'prediction': 'virginica'}
{'features': [5.4, 3.9, 1.7, 0.4]} {'prediction': 'setosa'}
{'features': [5.2, 3.5, 1.5, 0.2]} {'prediction': 'setosa'}
{'feat

Docker instructions:
1. To create the docker image: docker build -t iris-flask-app .
2. To run the docker container: docker run -p 5000:5000 iris-flask-app
3. To make predictions in the terminal: curl -X POST http://localhost:5000/predict -H "Content-Type: application/json" -d "{\"features\": [5.1, 3.5, 1.4, 0.2]}"
4. In order for the web app to be accessed go to http://localhost:5000/home
5. To use the web app in other devices connected to a common local network, find the corresponding IPv4 Address and use `http:"IP address":5000/home`.